xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -1165,6 +1170,25 @@ bool CDVDDemuxFFmpeg::SeekTime(double time, bool backwards, double* startpts)
 
   if (m_checkvideo)
   {
+    int elapsedMs = 0;
+    auto started = std::chrono::system_clock::now();
+
+    while (!IsVideoReady() && elapsedMs < 1000)
+    {
+      DemuxPacket *pkt = Read();
+      if (pkt)
+        CDVDDemuxUtils::FreeDemuxPacket(pkt);
+      else
+        Sleep(10);
+      m_pkt.result = -1;
+      av_packet_unref(&m_pkt.pkt);
+
+      elapsedMs = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now() - started).count();
+    }
+
+    if (elapsedMs >= 1000)
```

Comment:
Perhaps move this into the loop and get rid of the "elapsedMs < 1000" condition for the while statement above? Seems cleaner to me.

Reply:
will incorporate